### ISSUE TITLE
[JOSS] error running examples

### ISSUE THREAD

**Issue by matbesancon** [fsobral/RAFF.jl#11]
Run with Julia opened in the `examples/` folder:
```julia
julia> include("./runraff.jl")
[ Info: Precompiling RAFF [4aa82a78-ed18-41f9-aee6-9d73ba3a0b42]
ERROR: LoadError: MethodError: no method matching raff(::getfield(Main, Symbol("##3#4")), ::Array{Any,2}, ::Int64)
Closest candidates are:
  raff(::Function, ::Array{Float64,2}, ::Int64; kwargs...) at /home/mbesancon/.julia/dev/RAFF/src/RAFF.jl:516
  raff(::Function, ::Function, ::Array{Float64,2}, ::Int64; MAXMS, SEEDMS, initguess, ε, noutliers) at /home/mbesancon/.julia/dev/RAFF/src/RAFF.jl:386
Stacktrace:
 [1] top-level scope at /home/mbesancon/.julia/dev/RAFF/examples/runraff.jl:34
 [2] include at ./boot.jl:326 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1038
 [4] include(::Module, ::String) at ./sysimg.jl:29
 [5] include(::String) at ./client.jl:403
 [6] top-level scope at none:0
in expression starting at /home/mbesancon/.julia/dev/RAFF/examples/runraff.jl:9
```

There should be some explanation in the README what the `examples` and `scripts` folders are for, are they different? Should they be tested by Travis CI to ensure they stay up to date?

This issue is raised as part of https://github.com/openjournals/joss-reviews/issues/1385

**Comment by fsobral** [comment]
The `examples` directory came from old versions of `RAFF` and will be moved to `tests`, since some tests use its files.

The `script` directory is the place where we insert scripts for specific tests, conceptual examples and problem generation and visualization. They definitely should stay up-to-date. I'll work on that issue.

**Comment by matbesancon** [comment]
Ok great. When you move the `examples`, add to the documentation and README that examples can be found in the tests, and distinguish unit tests from these examples if you can

**Comment by fsobral** [comment]
Dear @matbesancon . I think we managed to organize the examples and scripts:

  - Inside `examples` we created some scripts to run small examples of `RAFF`. In particular, the examples form the tutorial were implemented.
  - All the test files inside `examples` were moved to `test`, since they were only used inside the unit tests.
  - The `script` directory has also been moved to `test` and an explanation of each file has been added to the README and documentation.

Thanks for the suggestion.
